app/controllers/items/history.js

```diff
@@ -0,0 +1,74 @@
+import detail from "./detail";
+import Ember from "ember";
+import _ from "lodash";
+
+export default detail.extend({
+  enableFooterActions: false,
+
+  groupedVersions() {
+    let versions = this.get("model.versions").toArray();
+
+    let results = _.map(versions, function(version) {
+      return _.omit(version.get("objectChanges"), [
+        "available_quantity",
+        "designated_quantity",
+        "on_hand_quantity",
+        "dispatched_quanitity",
+        "id",
+        "detail_type_id",
+        "created_at",
+        "received_at"
+      ]);
+    });
+    versions.forEach((version, index) =>
+      version.set("objectChanges", results[index])
+    );
+    return versions;
+  },
+
+  getGroupKey(isAction, action) {
+    const createdAt = moment(action.get("createdAt")).format("LL");
+    if (isAction) {
+      return `${action.get("user.fullName")}/${action.get(
+        "action"
+      )}/${createdAt}`;
+    } else {
+      return `${action.get("whodunnitName")}/${action.get(
+        "state"
+      )}/${createdAt}`;
+    }
+  },
+
+  groupingAndSortingActions(actionsAndVersions) {
+    const groups = _.reduce(
+      actionsAndVersions,
+      (results, action) => {
+        const createdAt = moment(action.get("createdAt")).format("LL");
+        const isItemAction = !!action.get("user.fullName");
+        const groupKey = this.getGroupKey(isItemAction, action);
+
+        results[groupKey] = results[groupKey] || {
+          key: groupKey,
+          type: isItemAction ? action.get("action").capitalize() : "Edited",
+          date: createdAt,
+          user: isItemAction
+            ? action.get("user.fullName")
+            : action.get("whodunnitName"),
```

Comment:
i see you're also doing similar logic here. So might be handy to have some helper methods like `getAuthor(change)` which you can use everywhere in your code. Those could have the ternary inside, which would clear up a lot of the code.

Since you don't actually know whether it's an inventory record or a version record, using a variable named "change" makes a bit more sense to me (more generic). Just a personal preference though